### PR TITLE
Speedup

### PR DESCRIPTION
**Pull request by michalsta**

### CODE REVIEW

File: src/features/population_generator_for_poland.py [hunk 1]
```diff
@@ -9,11 +9,31 @@
...
+def transform(subpopulation_df):
+    ret = []
+    probs = []
+
+    for row in subpopulation_df.to_dict(orient='records'):
+        fem_prob = row['female_probability']
+        male_prob = 1.0 - fem_prob
+        fem_prob *= row['total_probability']
+        male_prob *= row['total_probability']
+        ret.append((row['Age'], GENDERS[0]))
```
Comment (onomatopeia) [inline src/features/population_generator_for_poland.py:24]: GENDERS variable is unknown

File: src/features/population_generator_for_poland.py [hunk 2]
```diff
@@ -85,22 +111,28 @@ def _preprocess_household_headcount_ac(self) -> pd.DataFrame:
...
-        row = self.households_headcount_ac_df.iloc[idx]
+        #print(self.households_headcount_ac_df)
+        #idx = np.random.choice(self.households_headcount_ac_df.index.tolist(),
+        #                       p=self.households_headcount_ac_df['probability'])
+        #print(self.households_headcount_ac_df.index)
+        row = next(self.presampled_households) #self.households_headcount_ac_df.iloc[idx]
         return int(row['children']), int(row['adults'])
 
-    def _draw_from_subpopulation(self, subpopulation: pd.DataFrame, headcount: int, household_idx: int,
+    def _draw_from_subpopulation(self, subpopulation, headcount: int, household_idx: int,
```
Comment (onomatopeia) [inline src/features/population_generator_for_poland.py:114]: missing type of subpopulation

File: src/features/population_generator_for_poland.py [hunk 3]
```diff
@@ -85,22 +111,28 @@ def _preprocess_household_headcount_ac(self) -> pd.DataFrame:
...
+        #print(subpopulation)
 
         for _ in range(headcount):
-            idx = np.random.choice(subpopulation.index.tolist(), p=subpopulation['total_probability'])
-            row = subpopulation.iloc[idx]
-            age = row['Age']
-            gender = GENDERS[np.random.choice([0, 1], p=[row.female_probability, 1 - row.female_probability])]
+        #    print(subpopulation.index.tolist())
+        #    idx = np.random.choice(subpopulation.index.tolist(), p=subpopulation['total_probability'])
+        #    row = subpopulation.iloc[idx]
```
Comment (onomatopeia) [inline src/features/population_generator_for_poland.py]: please remove obsolete code

File: src/features/population_generator_for_poland.py [hunk 4]
```diff
@@ -85,22 +111,28 @@ def _preprocess_household_headcount_ac(self) -> pd.DataFrame:
...
+            row = next(subpopulation)
+            age = row[0]
+            gender = row[1] #GENDERS[np.random.choice([0, 1], p=[row.female_probability, 1 - row.female_probability])]
```
Comment (onomatopeia) [inline src/features/population_generator_for_poland.py:123]: please remove obsolete code

File: src/features/population_generator_for_poland.py [hunk 5]
```diff
@@ -9,11 +9,31 @@
 import numpy as np
 from tqdm import tqdm
 from xlrd import XLRDError
+import mocos_helper
 
 
 project_dir = Path(__file__).resolve().parents[2]
 poland_folder = project_dir / 'data' / 'processed' / 'poland'
 
+def transform(subpopulation_df):
```
Comment (onomatopeia) [inline src/features/population_generator_for_poland.py:15]: please add types info

File: src/features/population_generator_for_poland.py (hunk 1 above)
Comment (michalsta) [reply src/features/population_generator_for_poland.py:24]: Re-added, it got lost during merge.

**Comment by michalsta** [comment]
All right, I think I've fixed all mentioned issues.

**Comment by onomatopeia** [comment]
LGTM

**Pull request merged**